File: Python_Scripting_for_Computational_Science_Third_Edition/perl/hwGUI9_grid.pl
: # *-*-perl-*-*
  eval 'exec perl -w -S  $0 ${1+"$@"}' 
    if 0;  # if running under some shell

use Tk;
$main_win = MainWindow->new();
$top = $main_win->Frame(); 
$top->pack(-side => 'top');

# use grid to place widgets in 3x4 cells:
$font = "times 18 bold";
$hwtext = $top->Label(-text => "Hello, World!", -font => $font);
$hwtext->grid(-row => 0, -column => 0, -columnspan => 4, -pady => 20);

$top->Label(-text => 'The sine of ')->grid(-row => 1, -column => 0);
$r = 1.2;  # default
$r_entry = $top->Entry(-width => 6, -relief => 'sunken',
                       -textvariable => \$r);
$r_entry->grid(-row => 1, -column => 1);
$r_entry->bind('<Return>', \&comp_s);
sub comp_s { $s = sin($r); }

$top->Button(-text => " equals ", -relief => 'flat', 
             -command => \&comp_s)->grid(-row => 1, -column => 2);
$top->Label(-textvariable => \$s, -width => 18)
            ->grid(-row => 1, -column => 3);

$top->Button(-text => 'Goodbye, GUI World!', -command => \&exit,
             -background => 'yellow',  -foreground => 'blue')
             ->grid(-row => 2, -column => 0, -columnspan => 4,
                    -pady => 5, -sticky => 'ew');

$main_win->bind('<q>', \&exit);

MainLoop();

File: Python_Scripting_for_Computational_Science_Third_Edition/py/gui/hwGUI9_grid.py
#!/usr/bin/env python
from Tkinter import *
import math

root = Tk()              # root (main) window
top = Frame(root)        # create frame
top.pack(side='top')     # pack frame in main window

# use grid to place widgets in 3x4 cells:

font = 'times 18 bold'
hwtext = Label(top, text='Hello, World!', font=font)
hwtext.grid(row=0, column=0, columnspan=4, pady=20)

r_label = Label(top, text='The sine of')
r_label.grid(row=1, column=0)

r = StringVar() # variable to be attached to r_entry
r.set('1.2')    # default value
r_entry = Entry(top, width=6, textvariable=r)
r_entry.grid(row=1, column=1)

s = StringVar() # variable to be attached to s_label
def comp_s(event=None):
    s.set('%g' % math.sin(float(r.get()))) # construct string

r_entry.bind('<Return>', comp_s)

compute = Button(top, text=' equals ', command=comp_s, relief='flat')
compute.grid(row=1, column=2)

s_label = Label(top, textvariable=s, width=12)
s_label.grid(row=1, column=3)

# finally, make a quit button:
def quit(event=None):
    root.destroy()
quit_button = Button(top, text='Goodbye, GUI World!', command=quit,
                     background='yellow', foreground='blue')
quit_button.grid(row=2, column=0, columnspan=4, pady=5, sticky='ew')
root.bind('<q>', quit)

root.mainloop()
